core/node/node_framework/src/implementations/layers/web3_api/server/mod.rs

```diff
@@ -201,15 +206,28 @@ impl WiringLayer for Web3ServerLayer {
             number_updater: sealed_l2_block_handle.clone(),
             pool: updaters_pool,
         };
-        // Bridge addresses updater task must be started for ENs and only for ENs.
-        let bridge_addresses_updater_task =
-            input
-                .main_node_client
-                .map(|main_node_client| BridgeAddressesUpdaterTask {
-                    bridge_address_updater: bridge_addresses_handle.clone(),
-                    main_node_client: main_node_client.0,
-                    update_interval: self.optional_config.bridge_addresses_refresh_interval,
-                });
+
+        // In case it is an EN, the bridge addresses should be updated by fetching values from the main node.
+        // It is the main node, the bridge addresses need to be updated by querying the L1.
+
+        let bridge_addresses_updater_task = if let Some(main_node_client) = input.main_node_client {
```

Comment:
Let's also do the first `loop_iteration` right after task is created so we init API with correct bridge address.

Reply:
discussed in private, not needed